### PR TITLE
paraview: Fix build with DIY2.

### PR DESCRIPTION
**Pull request by rubendibattista**
Meanwhile we wait to fix python variant with #4045, #4707 was merged with the new version. Unfortunately `diy2` that is shipped with Paraview is patched w.r.t. the one present in the ports. So it needs to be compiled again from the source shipped with Paraview. Otherwise the build does not go on.

Moreover  the parallel build works. 

PS: Is there any chance we could extend build time for the CI workers in order to complete such long build jobs?

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

/cc @MarcusCalhoun-Lopez

### CODE REVIEW

File: science/paraview/Portfile [hunk 1]
```diff
@@ -111,7 +107,7 @@ configure.args-append \
     -DQT_HELP_GENERATOR=${qt_bins_dir}/qhelpgenerator \
     -DQT_XMLPATTERNS_EXECUTABLE=${qt_bins_dir}/xmlpatterns \
     -DVTK_USE_SYSTEM_LIBRARIES:BOOL=ON \
-    -DVTK_USE_SYSTEM_XDMF2:BOOL=OFF
+    -DVTK_USE_SYSTEM_XDMF2:BOOL=OFF \
```
Comment (pmetzger) [inline science/paraview/Portfile]: Adding a trailing \ here is definitely wrong.

**Comment by pmetzger** [comment]
@jjstickel? @dstrubbe?

**Comment by jjstickel** [comment]
@rubendibattista can you summarize where these fixes/updates are at? I'm a little confused by this PR. If merged, does it use bundled diy2 instead of macports diy2? If macports diy2 is installed, does it cause problems? Are you suggesting to have this PR merged, then  #4045?

**Comment by rubendibattista** [comment]
@jjstickel Paraview does not build with Macports version of diy2 when also MPI is enabled, at least on my machine (and if I remember well also on Linux, it needs the vendored version shipped by Paraview for what I understand). 

If anyone can confirm this would be great. Using the vendored version of diy2 makes it build on my machine.

So yes, I suggest to merge this. For what concerns #4045 I'm still trying to find a solution...

**Comment by rubendibattista** [comment]
>can you summarize where these fixes/updates are at?

This basically uses the vendored `diy2` version shipped with ParaView when the MPI is enabled.

**Comment by jjstickel** [comment]
OK, thanks for the explanations. This PR looks good to me.

Have you tried this installing paraview (with this patch) while macports diy2 is also installed? I don't want to hold up merging, but it would be good to know if there is a conflict.

**Comment by rubendibattista** [comment]
Ok this is not needed anymore, I fixed also the python build in #4045. So we can merge that one directly that also incorporates this one.
